qiskit/visualization/utils.py

```diff
@@ -208,5 +175,150 @@ def _get_gate_span(qregs, instruction):
 
     if instruction.cargs:
         return qregs[min_index:]
+    if instruction.condition and isinstance(instruction.condition[0], ClassicalRegister):
+        return qregs[min_index:]
 
     return qregs[min_index:max_index + 1]
+
+
+def _any_crossover(qregs, node, nodes):
+    """Return True .IFF. 'node' crosses over any in 'nodes'
+    qiskit-terra #2802
+    """
+    gate_span = _get_gate_span(qregs, node)
+    all_indices = []
+    for check_node in nodes:
+        if check_node != node:
+            all_indices += _get_gate_span(qregs, check_node)
+    return any(i in gate_span for i in all_indices)
+
+
+class Justification(Enum):
```

Comment:
I'm not really sure this enum is necessary as it is only used once to set the justification. You could just pass the string that has already [been validated](https://github.com/Qiskit/qiskit-terra/blob/35abeb9da0f8f975931a8f8bd281a64d6df0ba82/qiskit/visualization/utils.py#L87) to be either `'left'` or `'right'`.

Reply:
Done.